apollo-router/src/uplink/license_enforcement.rs

```diff
@@ -135,6 +138,43 @@ impl ParsedLinkSpec {
             })
     }
 
+    fn from_join_directive_args(
+        args: &[(Name, Node<ast::Value>)],
+    ) -> Option<Result<Self, url::ParseError>> {
+        let url_string = args
+            .iter()
+            .find(|(name, _)| name == &Name::new_unchecked(LINK_URL_ARGUMENT))
+            .and_then(|(_, value)| value.as_str());
+
+        let parsed_url = Url::parse(url_string.unwrap_or_default()).ok()?;
+
+        let mut segments = parsed_url.path_segments()?;
+        let spec_name = segments.next()?.to_string();
+        let spec_url = format!(
+            "{}://{}/{}",
+            parsed_url.scheme(),
+            parsed_url.host()?,
+            spec_name
+        );
+        let version_string = segments.next()?.strip_prefix('v')?;
+        let parsed_version =
+            semver::Version::parse(format!("{}.0", &version_string).as_str()).ok()?;
```

Comment:
Is appending this `.0` always correct? Is it enforced that the version string in the URL is always major.minor before we get here?

Reply:
good question, but yeah — the link spec doesn't support patch versions: https://specs.apollo.dev/link/v1.0/#@link.url